wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/FindId.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.languageserver;
+
+import com.google.common.collect.ImmutableSet;
+import java.util.HashSet;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.regex.Pattern;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.languageserver.RegistryContainer.Registry;
+
+/** Utility class that simplifies finding language server id. */
+@Singleton
+class FindId {
+  private final Registry<Set<Pattern>> patterns;
+
+  @Inject
+  FindId(RegistryContainer registryContainer) {
+    this.patterns = registryContainer.patternRegistry;
```

Comment:
The class in the example above has 54 lines (including license), and there is only place where mentioned variable is used, that does not seem hard.

You see, probably you do not bother yourself with such trifles but in Java there is quite a difference between term variable and field, in your case you probably meant exactly field. So when we are talking about final fields that are initialized by injected or with the help of injected components I don't think it is an anti-pattern.

Reply:
How do I do a search for all references to registryContaine.patternRegistry? Since the static field is copied, I don't really know how.
Btw. You may safely assume that I am familiar with the ins and outs of Java, having approximately 20 years experience in it.